---
title: "Exp160_Final"
author: "ACY"
date: "2024-11-28"
output: html_document
---
## Used module R/4.4.0


```{r, warning=FALSE, message=FALSE, warning=FALSE, echo=F}
set.seed(1234)
#graphics.off()
#rm(list=ls())
#knitr::opts_chunk$set(fig.width=8, fig.height=6, dpi=300,echo=FALSE, warning=FALSE, message=FALSE)
#knitr::opts_chunk$set(dev.args=list(bg="transparent"))
#ROOT_DIR<-"~/Fred Hutchinson Cancer Research Center/GVHD - Hill Furlan Collaboration - General"
setwd("/fh/fast/hill_g/Albert/scSeq_Analyses/BD_Rhapsody_Motoko_ISC_2024/Exp160/data")
ROOT_DIR<-"/fh/fast/hill_g/Albert/scSeq_Analyses/BD_Rhapsody_Motoko_ISC_2024/"
stem<-"Exp160"
DATA_DIR <- file.path(ROOT_DIR, stem, "data")      # SPECIFY HERE
RES_DIR  <- file.path(ROOT_DIR, stem, "res")     # SPECIFY HERE
RMD_DIR  <- file.path(ROOT_DIR, stem, "rmd")     # SPECIFY HERE
CDS_DIR <- file.path(ROOT_DIR, stem, "cds")
FIG_DIR <- file.path(ROOT_DIR,  stem, "figs")

ENR_DIR<-file.path(ROOT_DIR, "Bioinformatics Resources", "MSigDB")
SCRIPT_DIR<-file.path(ROOT_DIR, "develop", "SfScripts", "m3")

suppressPackageStartupMessages({
  library(Seurat)
  library(SeuratWrappers) # Needed to run monocle3; install as follows: devtools::install_github('satijalab/seurat-wrappers')
  library(scCustomize) # Needed to run SetUP.R (/fh/fast/hill_g/Albert/scSeq_Analyses/Setup Files)
    #Error: ERROR: no permission to install to directory ‘/home/ayeh/R/x86_64-pc-linux-gnu-library/4.3’
  library(SCpubr) 
  library(monocle3)
  library(m3addon) #plugin for monocle 
  #options(repos = c(CRAN = "https://cloud.r-project.org")) 
  #devtools::install_github('scfurl/m3addon')
  # ml HDF5/1.14.3-gompi-2023b

  library(ggplot2)
  library(ggplot2)
  library(patchwork)
  library(RColorBrewer)
  library(tidyverse)
  library(scCustomize)
  library(ArchR)
  library(viridis)
  library(ggpubr)
  library(Matrix)
  # devtools::install_github("immunogenomics/presto")
  
  #dyn.load('/app/software/ArrayFire/3.8.1/lib64/libaf.so.3')
  #library(RcppArrayFire)
  #library(viewmastR) # https://github.com/scfurl/viewmaster
})

#hg38<-getBSgenome("BSgenome.Hsapiens.UCSC.hg38" )

xfun::pkg_load2(c('base64enc', 'htmltools', 'mime'))
#install.packages("tidyverse")
#devtools::install_github("waltno/viewmaster")

SFtheme<-theme_bw(base_size=14) + 
  theme(panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
    panel.grid.minor = element_blank(), 
    panel.grid.major = element_blank(),
    legend.background = element_rect(fill = "transparent"),
    legend.box.background = element_rect(fill = "transparent"), 
    legend.key = element_rect(fill = "transparent", colour = NA),
    plot.background = element_rect(fill = "transparent",colour = NA))
theme_set(SFtheme)


clus_cols12 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78")
clus_cols16 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78", 
                 "#98df8a", "#ff9896", "#c5b0d5", "#c49c94")
clus_cols24 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78", 
                 "#98df8a", "#ff9896", "#c5b0d5", "#c49c94", "#f7b6d2", "#c7c7c7", 
                 "#dbdb8d", "#9edae5", "#ad494a", "#8c6d31", "#e7ba52", "#17becf")
clus_cols36 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78",
                 "#98df8a", "#ff9896", "#c5b0d5", "#c49c94", "#f7b6d2", "#c7c7c7", 
                 "#dbdb8d", "#9edae5", "#ad494a", "#8c6d31", "#e7ba52", "#17becf",
                 "#1f78b4", "#33a02c", "#fb9a99", "#e31a1c", "#fdbf6f", "#cab2d6",
                 "#6a3d9a", "#ff8c00", "#b15928", "#41ab5d", "#f03b20", "#807dba")

#pie(rep(1, length(col)), col = col)
datestamp<-"241108"

rna_cols<-paletteContinuous(n=8)[c(1:3, 6:8)]

```

```{r - loads RSEC data, echo=FALSE, include = TRUE}
## this uses RSEC
f<-list.files(file.path(DATA_DIR, paste0("cart", 1:2)), full.names = T)[grep("RSEC_MolsPerCell", list.files(file.path(DATA_DIR, paste0("cart", 1:2))))] #This line lists all files in two directories, cart1 and cart2, inside DATA_DIR.
files<-f[c(2,4)]

dat<-lapply(files, function(file){
t(as.matrix(data.table::fread(file), rownames=1))}) # For each file in files, it reads the file into a matrix using data.table::fread. The t() function transposes each matrix, making rows into columns and columns into rows. dat will contain two transposed matrices from the selected files.

rn<-intersect(rownames(dat[[1]]), rownames(dat[[2]])) # rn is a list of row names common to both matrices.
rn<-sapply(strsplit(rn, "\\|"), "[[", 1)  # The strsplit(rn, "\\|") splits each row name by the "|" delimiter. sapply(..., "[[", 1) extracts the first part before the delimiter for each name.

dat2<-lapply(dat, function(mat){
  rownames(mat)<-sapply(strsplit(rownames(mat), "\\|"), "[[", 1)
  mat[rn,]
}) #This modifies the row names of each matrix by keeping only the first part (before the "|"). Each matrix is then subsetted to include only the rows that match the names in rn, creating a new list dat2 with the processed matrices.

colnames(dat2[[1]])<-paste0("cart1_", colnames(dat2[[1]]))
colnames(dat2[[2]])<-paste0("cart2_", colnames(dat2[[2]]))
mat<-Matrix(do.call(cbind, dat2), sparse=T) # The two matrices in dat2 are combined column-wise (i.e., bound together as columns) using do.call(cbind, dat2). The Matrix() function with sparse=T creates a sparse matrix from the combined data, saving memory by storing only non-zero entries.
colnames(mat)<-make.unique(colnames(mat)) # This line ensures that all column names in the combined matrix mat are unique by appending numbers where duplicate names are found.

rmat<-mat[5:dim(mat)[1],] # rmat contains the main GEX data starting from the 8th row.
pmat<-mat[1:4,] # pmat contains the first 7 rows and holds the ADT data

seu<-CreateSeuratObject(counts = rmat)
seu$ADT<-CreateAssayObject(counts = pmat[,match(colnames(pmat), Cells(seu))])

#DefaultAssay(seu)
## ADT Features:
# [1] "CD326"        
# [5] "CD45.2"             "H-2Kb"              "I-A-I-E"  

seu <- NormalizeData(seu, normalization.method = "LogNormalize", scale.factor = 10000)
seu <- FindVariableFeatures(seu, selection.method = "vst", nfeatures = 3000)
seu <- ScaleData(seu)
seu <- RunPCA(seu, features = VariableFeatures(object = seu), npcs = 50)
#ElbowPlot(seu, 50)
seu <- FindNeighbors(seu, dims = 1:40)
seu <- FindClusters(seu, resolution = 0.5)
seu <- RunUMAP(seu, dims = 1:40)

DimPlot(seu)
seu$cart<-strsplit(Cells(seu), "_") %>% sapply("[[", 1)
seu[["percent.mt"]] <- PercentageFeatureSet(seu, pattern = "^mt-")
seu$logUMI<-log10(seu$nCount_RNA)
seu$log_RNA<-log10(seu$nCount_RNA)

## For QC presentation ##
VlnPlot(seu, features = c("nFeature_RNA", "log_RNA", "logUMI", "percent.mt"), group.by = "cart", ncol = 4, pt.size = 0)
DimPlot(seu, group.by = "cart")
FeaturePlot_scCustom(seu, features = "percent.mt")
####

## Now filter by RNA count
DefaultAssay(seu_sub)<-"RNA"

#seu_sub <- subset(seu, subset = log_RNA > 3.25 & log_RNA < 4.25 & percent.mt < 40) 
seu_sub <- subset(seu, subset = nCount_RNA > 1000 & nCount_RNA < 50000 & percent.mt < 25) # This gives us 8,032 samples after filtering (from 15,288 before0)
seu_sub <- NormalizeData(seu_sub, normalization.method = "LogNormalize", scale.factor = 10000)
seu_sub <- FindVariableFeatures(seu_sub, selection.method = "vst", nfeatures = 3000)
seu_sub <- ScaleData(seu_sub)
seu_sub <- RunPCA(seu_sub, features = VariableFeatures(object = seu_sub), npcs = 50)
#ElbowPlot(seu_sub, 50)
seu_sub <- FindNeighbors(seu_sub, dims = 1:40)
seu_sub <- FindClusters(seu_sub, resolution = 0.15)
seu_sub <- RunUMAP(seu_sub, dims = 1:40)

## Data output for QC ##
DimPlot(seu_sub)
VlnPlot(seu_sub, features = c("nFeature_RNA", "log_RNA", "logUMI", "percent.mt"), group.by = "cart", ncol = 4, pt.size = 0)
DimPlot(seu_sub, group.by = "cart")
FeaturePlot_scCustom(seu_sub, features = "percent.mt")
#####

## Normalize ADT Data ##
DefaultAssay(seu_sub)<-"ADT"
seu_sub <- NormalizeData(seu_sub, normalization.method = "CLR")
seu_sub <- ScaleData(seu_sub)

DimPlot(seu_sub, cols = clus_cols36)

#### Temporary Save Point Here ####
#The RDS file below has removed multiplets / those without proper hashtags, as well as filter by RNA counts and mito genes.  
#Also, 4 groups (female IFNy +/- and male IFNy +/-) have been added
saveRDS(seu_sub, file.path(CDS_DIR, "241108v2_processed.RDS"))
####
```


```{r - analysis of clusters, echo=FALSE, include=TRUE}
seu<-readRDS(file.path(CDS_DIR, "241108v2_processed.RDS"))

pdf(file.path(FIG_DIR, paste0("[Final] - Initial Clustering, res = 0.15", ".pdf")), width=6, height=6)
DimPlot(seu, cols = clus_cols36)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Initial Clustering by cart", ".pdf")), width=6, height=6)
DimPlot(seu,  group.by = "cart")
dev.off()


FeaturePlot_scCustom(seu, features = "Mki67")
FeaturePlot_scCustom(seu, features = "percent.mt")

DefaultAssay(seu)<-"RNA"
markers_1_vs_5 <- FindMarkers(seu, ident.1 = 1, ident.2 = 5) #Helped identify Chgb and Cpe = enteroendocrine cells

pdf(file.path(FIG_DIR, paste0("[Final] - Chromogranin B", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Chgb")
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Muc2", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Muc2") # Goblet/Paneth
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Trpm5", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Trpm5") # Tufts
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Lyz1", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Lyz1") # Lysozyme (Paneth)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Mmp7", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Mmp7") # Mmp7 (Paneth)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Pou2f3", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Pou2f3") # Pou2f3 (Tufts)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Anpep", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Anpep") # Aminopeptidase N (Enterocyte)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Epcam", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Epcam") # EPCAM
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Lgr5", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Lgr5") # ISC
dev.off()


FeaturePlot_scCustom(seu, features = "Bmi1") # ISC
FeaturePlot_scCustom(seu, features = "Ephb2") # ISC
markers_1_vs_4 <- FindMarkers(seu, ident.1 = 1, ident.2 = 4) #Helped identify Chgb and Cpe = enteroendocrine cells
FeaturePlot_scCustom(seu, features = "Ccl25") # EPCAM

pdf(file.path(FIG_DIR, paste0("[Final] - Markers", ".pdf")), width=6, height=6)
DotPlot(seu, features = c("Chgb", "Cpe", "Muc2", "Lyz1", "Mmp7", "Trpm5", "Pou2f3", "Anpep", "Epcam","Lgr5","Ccl25")) + RotatedAxis()
dev.off()

```

```{r - expression markers, echo=FALSE, include=TRUE}


# [1] "CD326"        
# [5] "CD45.2"             "H-2Kb"              "I-A-I-E"  

DefaultAssay(seu)<-"ADT"

my_comparisons <- list( c("0","1"),c("0","2"),c("0","3"),c("0","4"),c("0","5"),c("0","6")) 

pdf(file.path(FIG_DIR, paste0("[Final] - I-A-I-E", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "I-A-I-E", max.cutoff = "q99") # Tufts
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - I-A-I-E (violin)" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("I-A-I-E"), group.by="seurat_clusters", col = clus_cols36,  pt.size = 0) + stat_summary(fun = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 12)
dev.off()


pdf(file.path(FIG_DIR, paste0("[Final] - H-2Kb", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "H-2Kb", max.cutoff = "q99") # Tufts
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H-2Kb (violin)" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("H-2Kb"), group.by="seurat_clusters", col = clus_cols36, pt.size = 0) + stat_summary(fun = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 14)
dev.off()


```

```{r - GSEA, echo=FALSE, include=FALSE}

#========== Figure 5E GSEA ==========#
#---- GSEA analysis using fsgea ----#
#https://crazyhottommy.github.io/scRNA-seq-workshop-Fall-2019/scRNAseq_workshop_3.html
library(presto)
library(dplyr)
library(ggplot2)
library(tibble)
library(fgsea)
library(msigdbr)

seu.genes <- wilcoxauc(seu, 'seurat_clusters')

# for GSEA, we need the information of all genes, Seurat is just too slow if we test
# all 20,000 genes. instead let's try presto which performs a fast Wilcoxon rank sum test 

#---- Ontology Gene Sets ----#
m_t2g <- msigdbr(species = "Mus musculus", category = "C5") %>% dplyr::select(gs_name, gene_symbol) 
fgsea_sets<- m_t2g %>% split(x = .$gene_symbol, f = .$gs_name) #Collects names of the genes of interest

#----- Comparing gene sets -----#
pdf(file.path(FIG_DIR, paste0("[Final] - GOCC_MHC_CLASS_II_PROTEIN_COMPLEX", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "GOCC_MHC_CLASS_II_PROTEIN_COMPLEX",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "MHC_CLASS_II_PROTEIN_COMPLEX")
FeaturePlot_scCustom(seu, features = "MHC_CLASS_II_PROTEIN_COMPLEX1", label = F, label.size = 2, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - GOCC_MHC_CLASS_II_PROTEIN_COMPLEX violin", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("MHC_CLASS_II_PROTEIN_COMPLEX1"), group.by="seurat_clusters", col = clus_cols36,  pt.size = 0) + stat_summary(fun = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 5)
dev.off()


pdf(file.path(FIG_DIR, paste0("[Final] - GOCC_MHC_CLASS_I_PROTEIN_COMPLEX", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "GOCC_MHC_CLASS_I_PROTEIN_COMPLEX",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "MHC_CLASS_I_PROTEIN_COMPLEX")
FeaturePlot_scCustom(seu, features = "MHC_CLASS_I_PROTEIN_COMPLEX1", label = F, label.size = 2, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - GOCC_MHC_CLASS_I_PROTEIN_COMPLEX violin", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("MHC_CLASS_I_PROTEIN_COMPLEX1"), group.by="seurat_clusters", col = clus_cols36,  pt.size = 0) + stat_summary(fun = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 2.5)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - GOBP_ANTIGEN_PROCESSING_AND_PRESENTATION", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "GOBP_ANTIGEN_PROCESSING_AND_PRESENTATION",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "ANTIGEN_PROCESSING_AND_PRESENTATION")
FeaturePlot_scCustom(seu, features = "ANTIGEN_PROCESSING_AND_PRESENTATION1", label = F, label.size = 2, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - GOBP_ANTIGEN_PROCESSING_AND_PRESENTATION violin", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("ANTIGEN_PROCESSING_AND_PRESENTATION1"), group.by="seurat_clusters", col = clus_cols36,  pt.size = 0) + stat_summary(fun = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 1)
dev.off()


DefaultAssay(seu)<-"RNA"
pdf(file.path(FIG_DIR, paste0("[Final] - Markers GSEA", ".pdf")), width=4, height=6)
DotPlot(seu, features = c("MHC_CLASS_II_PROTEIN_COMPLEX1", "MHC_CLASS_I_PROTEIN_COMPLEX1", "ANTIGEN_PROCESSING_AND_PRESENTATION1")) + 
  scale_x_discrete(labels = function(x) substr(x, 1, 15)) + 
  RotatedAxis()
dev.off()

DefaultAssay(seu)<-"ADT"
pdf(file.path(FIG_DIR, paste0("[Final] - Markers ADT", ".pdf")), width=4, height=6)
DotPlot(seu, features = c("I-A-I-E", "H-2Kb")) + RotatedAxis()
dev.off()



```

```{r - custom gene sets, echo=FALSE, include=FALSE}

#Gene sets:
#Grun D et al Nature 2015 (PMID: 26287467)
#Haber AL et al Nature 2017 (PMID: 29144463)
#Biton M et at. Cell 2018 (PMID: 30392957)
#Hadjsombati MS et al. (Tuft cell) Immunity 2018 (PMID: 30021144)
#Wang et al. JEM 2019 (PMID: 31753849)
#Zhao Y et al. Mucosal Immunol 2020 (PMID: 31900405)
#Yin J et al. Cell Discovery 2023 (https://www.nature.com/articles/s41421-023-00578-4)

#Stem cell: 	Lgr5, Olfm4, Slc12a2, Ascl2, Axin2, Gkn3
#Transit-amplifying cells (TA):	 Olfm4, Scl12a2, (nusap1 (found in human))
#Goblet:	Muc2, Agr2, Clca3, Tff3, Ang4, Defa31
#Tuft:		Dclk1, Trpm5, Gnat3 (α-gustducin), CD24a, Krt8, Krt18, Gfi1b, Il25, Ptgs1, Alox5, Ptprc, Chat , Rac2 (specific)
#Paneth:	Mptx1 (specific), Ang4, Defa31, Defa17, Defa24, Defa22, Lyz1, Itln1 (Paneth in mouse)
#Enteroendocrine: Reg4 (specific), Chga, Chgb, Tph1, Tac1, Neuro3, 
#Enterocytes: Alpi, Apoa1, Apoa4, Fabp1, 
#Partial_IEC: Top2a, Apol10a, Car4 (carbonic anhydrase IV), and Ube2c (a precursor intestinal cell marker), and these cells belonged to a stem cell cluster that has acquired partial IECs function


custom_gene_sets <- list(
  "Stem_Cell" = c("Lgr5", "Olfm4", "Slc12a2", "Ascl2", "Axin2", "Gkn3"),
  "Transit_Amplifying_Cells" = c("Olfm4", "Scl12a2"), # nusap1 found in human
  "Goblet" = c("Muc2", "Agr2", "Clca3", "Tff3", "Ang4", "Defa31"),
  "Tuft" = c("Dclk1", "Trpm5", "Gnat3", "CD24a", "Krt8", "Krt18", "Gfi1b", "Il25", "Ptgs1", "Alox5", "Ptprc", "Chat", "Rac2"), # Gnat3 is α-gustducin, Rac2 specific
  "Paneth" = c("Mptx1", "Ang4", "Defa31", "Defa17", "Defa24", "Defa22", "Lyz1", "Itln1"), # Mptx1 specific, Itln1 Paneth in mouse
  "Enteroendocrine" = c("Reg4", "Chga", "Chgb", "Tph1", "Tac1", "Neuro3"), # Reg4 specific
  "Enterocytes" = c("Alpi", "Apoa1", "Apoa4", "Fabp1"),
  "Partial_IEC" = c("Top2a","Apol10a","Car4","Ube2c")
)

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Stem_Cell", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Stem_Cell
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Stem_Cell")
FeaturePlot_scCustom(seu, features = "Stem_Cell1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Transit_Amplifying_Cells", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Transit_Amplifying_Cells
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Transit_Amplifying_Cells")
FeaturePlot_scCustom(seu, features = "Transit_Amplifying_Cells1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Goblet", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Goblet
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Goblet")
FeaturePlot_scCustom(seu, features = "Goblet1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Tuft", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Tuft
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Tuft")
FeaturePlot_scCustom(seu, features = "Tuft1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Paneth", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Paneth
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Paneth")
FeaturePlot_scCustom(seu, features = "Paneth1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Enteroendocrine", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Enteroendocrine
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Enteroendocrine")
FeaturePlot_scCustom(seu, features = "Enteroendocrine1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()


pdf(file.path(FIG_DIR, paste0("[Final] - Custom Enterocytes", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Enterocytes
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Enterocytes")
FeaturePlot_scCustom(seu, features = "Enterocytes1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Custom Partial_IEC", ".pdf")), width=6, height=6)
bac<-custom_gene_sets$Partial_IEC
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "Partial_IEC")
FeaturePlot_scCustom(seu, features = "Partial_IEC1", label = F, label.size = 1, pt.size = 1, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)
dev.off()

```

# Pseudotime Trajectory
```{r pseudotime}

library(GenomicRanges)
cds <- as.cell_data_set(seu)
cds <- cluster_cells(cds, resolution=1e-3)

cds<-learn_graph(cds, use_partition = T, close_loop = T)
cds@clusters@listData[["UMAP"]][["clusters"]]<-clusters(cds)


cds<- order_cells(cds)
plot_cells(cds, color_cells_by = "pseudotime", show_trajectory_graph = T)

seu$pseudotime<-pseudotime(cds)

pdf(file.path(FIG_DIR, paste0("[Final] - Pseudotime", ".pdf")), width=6, height=6)
FeaturePlot(seu, features = "pseudotime")+scale_color_gradientn(colors = viridis::plasma(n=10))
dev.off()



#### Old Stuff
cds<-seurat_to_monocle3(seu)
#troubleshooting
cds@int_colData@listData[["reducedDims"]]@listData<-cds@int_colData@listData[["reducedDims"]]@listData[1]

plot_cells(cds, genes = c("Sell", "Lef1"), scale_to_range = F)
cds<-cluster_cells(cds, resolution = 0.001)
plot_cells(cds)

cds@clusters@listData[["UMAP"]][["clusters"]]<-clusters(cds)
cds<-learn_graph(cds, use_partition = T, close_loop = T)
cds<- order_cells(cds)
plot_cells(cds, color_cells_by = "pseudotime", show_trajectory_graph = T)

seu$pseudotime<-pseudotime(cds)
pdf("../figs/cd4_pseudotime.pdf", width = 5, height = 4)
FeaturePlot(seu, features = "pseudotime")+scale_color_gradientn(colors = viridis::plasma(n=10))&NoAxes()
dev.off()
```