## Used module R/4.4.0 
---
title: "Exp649_Final"
author: ACY
output: html_myeebook
editor_options: 
  chunk_output_type: console
---

```{r, warning=FALSE, message=FALSE, warning=FALSE, echo=F}
set.seed(1234)
#graphics.off()
#rm(list=ls())
#knitr::opts_chunk$set(fig.width=8, fig.height=6, dpi=300,echo=FALSE, warning=FALSE, message=FALSE)
#knitr::opts_chunk$set(dev.args=list(bg="transparent"))
#ROOT_DIR<-"~/Fred Hutchinson Cancer Research Center/GVHD - Hill Furlan Collaboration - General"
setwd("/fh/fast/hill_g/Albert/scSeq_Analyses/BD_Rhapsody_Motoko_ISC_2024/Exp649/data")
ROOT_DIR<-"/fh/fast/hill_g/Albert/scSeq_Analyses/BD_Rhapsody_Motoko_ISC_2024"
stem<-"Exp649"
DATA_DIR <- file.path(ROOT_DIR, stem, "data")      # SPECIFY HERE
RES_DIR  <- file.path(ROOT_DIR, stem, "res")     # SPECIFY HERE
RMD_DIR  <- file.path(ROOT_DIR, stem, "rmd")     # SPECIFY HERE
CDS_DIR <- file.path(ROOT_DIR, stem, "cds")
FIG_DIR <- file.path(ROOT_DIR,  stem, "figs")

ENR_DIR<-file.path(ROOT_DIR, "Bioinformatics Resources", "MSigDB")
SCRIPT_DIR<-file.path(ROOT_DIR, "develop", "SfScripts", "m3")

suppressPackageStartupMessages({
  library(Seurat)
  library(SeuratWrappers) # Needed to run monocle3; install as follows: devtools::install_github('satijalab/seurat-wrappers')
  library(scCustomize) # Needed to run SetUP.R (/fh/fast/hill_g/Albert/scSeq_Analyses/Setup Files)
    #Error: ERROR: no permission to install to directory ‘/home/ayeh/R/x86_64-pc-linux-gnu-library/4.3’
  library(SCpubr) 
  #library(monocle3)
  #library(m3addon) #plugin for monocle
  library(ggplot2)
  library(ggplot2)
  library(patchwork)
  library(RColorBrewer)
  library(tidyverse)
  library(scCustomize)
  library(ArchR)
  library(viridis)
  library(ggpubr)
  library(Matrix)
  # devtools::install_github("immunogenomics/presto")
  
  #dyn.load('/app/software/ArrayFire/3.8.1/lib64/libaf.so.3')
  #library(RcppArrayFire)
  #library(viewmastR) # https://github.com/scfurl/viewmaster
})

#hg38<-getBSgenome("BSgenome.Hsapiens.UCSC.hg38" )

xfun::pkg_load2(c('base64enc', 'htmltools', 'mime'))
#install.packages("tidyverse")
#devtools::install_github("waltno/viewmaster")

SFtheme<-theme_bw(base_size=14) + 
  theme(panel.background = element_rect(fill = "transparent",colour = NA), # or theme_blank()
    panel.grid.minor = element_blank(), 
    panel.grid.major = element_blank(),
    legend.background = element_rect(fill = "transparent"),
    legend.box.background = element_rect(fill = "transparent"), 
    legend.key = element_rect(fill = "transparent", colour = NA),
    plot.background = element_rect(fill = "transparent",colour = NA))
theme_set(SFtheme)


clus_cols12 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78")
clus_cols16 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78", 
                 "#98df8a", "#ff9896", "#c5b0d5", "#c49c94")
clus_cols24 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78", 
                 "#98df8a", "#ff9896", "#c5b0d5", "#c49c94", "#f7b6d2", "#c7c7c7", 
                 "#dbdb8d", "#9edae5", "#ad494a", "#8c6d31", "#e7ba52", "#17becf")
clus_cols36 <- c("#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", 
                 "#e377c2", "#7f7f7f", "#bcbd22", "#17becf", "#aec7e8", "#ffbb78",
                 "#98df8a", "#ff9896", "#c5b0d5", "#c49c94", "#f7b6d2", "#c7c7c7", 
                 "#dbdb8d", "#9edae5", "#ad494a", "#8c6d31", "#e7ba52", "#17becf",
                 "#1f78b4", "#33a02c", "#fb9a99", "#e31a1c", "#fdbf6f", "#cab2d6",
                 "#6a3d9a", "#ff8c00", "#b15928", "#41ab5d", "#f03b20", "#807dba")

#pie(rep(1, length(col)), col = col)
datestamp<-"241108"

rna_cols<-paletteContinuous(n=8)[c(1:3, 6:8)]

```


```{r - loads RSEC data, echo=FALSE, include = TRUE}
## this uses RSEC
f<-list.files(file.path(DATA_DIR, paste0("cart", 1:2)), full.names = T)[grep("RSEC_MolsPerCell", list.files(file.path(DATA_DIR, paste0("cart", 1:2))))] #This line lists all files in two directories, cart1 and cart2, inside DATA_DIR.
files<-f[c(2,4)]

dat<-lapply(files, function(file){
t(as.matrix(data.table::fread(file), rownames=1))}) # For each file in files, it reads the file into a matrix using data.table::fread. The t() function transposes each matrix, making rows into columns and columns into rows. dat will contain two transposed matrices from the selected files.

rn<-intersect(rownames(dat[[1]]), rownames(dat[[2]])) # rn is a list of row names common to both matrices.
rn<-sapply(strsplit(rn, "\\|"), "[[", 1)  # The strsplit(rn, "\\|") splits each row name by the "|" delimiter. sapply(..., "[[", 1) extracts the first part before the delimiter for each name.

dat2<-lapply(dat, function(mat){
  rownames(mat)<-sapply(strsplit(rownames(mat), "\\|"), "[[", 1)
  mat[rn,]
}) #This modifies the row names of each matrix by keeping only the first part (before the "|"). Each matrix is then subsetted to include only the rows that match the names in rn, creating a new list dat2 with the processed matrices.

colnames(dat2[[1]])<-paste0("cart1_", colnames(dat2[[1]]))
colnames(dat2[[2]])<-paste0("cart2_", colnames(dat2[[2]]))
mat<-Matrix(do.call(cbind, dat2), sparse=T) # The two matrices in dat2 are combined column-wise (i.e., bound together as columns) using do.call(cbind, dat2). The Matrix() function with sparse=T creates a sparse matrix from the combined data, saving memory by storing only non-zero entries.
colnames(mat)<-make.unique(colnames(mat)) # This line ensures that all column names in the combined matrix mat are unique by appending numbers where duplicate names are found.

rmat<-mat[8:dim(mat)[1],] # rmat contains the main GEX data starting from the 8th row.
pmat<-mat[1:7,] # pmat contains the first 7 rows and holds the ADT data

seu<-CreateSeuratObject(counts = rmat)
seu$ADT<-CreateAssayObject(counts = pmat[,match(colnames(pmat), Cells(seu))])

#DefaultAssay(seu)
## ADT Features:
# [1] "CD326:G8.8-AMM2281" "CD326:G8.8-AMM2294" "CD326:G8.8-AMM2295" "CD326"             
# [5] "CD45.2"             "H-2Kb"              "I-A-I-E"  

seu <- NormalizeData(seu, normalization.method = "LogNormalize", scale.factor = 10000)
seu <- FindVariableFeatures(seu, selection.method = "vst", nfeatures = 3000)
seu <- ScaleData(seu)
seu <- RunPCA(seu, features = VariableFeatures(object = seu), npcs = 50)
#ElbowPlot(seu, 50)
seu <- FindNeighbors(seu, dims = 1:40)
seu <- FindClusters(seu, resolution = 0.5)
seu <- RunUMAP(seu, dims = 1:40)

DimPlot(seu)
seu$cart<-strsplit(Cells(seu), "_") %>% sapply("[[", 1)
seu[["percent.mt"]] <- PercentageFeatureSet(seu, pattern = "^mt-")
seu$logUMI<-log10(seu$nCount_RNA)
seu$log_RNA<-log10(seu$nCount_RNA)


## For QC presentation ##
VlnPlot(seu, features = c("nFeature_RNA", "log_RNA", "logUMI", "percent.mt"), group.by = "cart", ncol = 4, pt.size = 0)
DimPlot(seu, group.by = "cart")
FeaturePlot_scCustom(seu, features = "percent.mt")
####


## Now filter by RNA count

#seu_sub <- subset(seu, subset = log_RNA > 3.25 & log_RNA < 4.25 & percent.mt < 40) 
seu_sub <- subset(seu, subset = nCount_RNA > 1000 & nCount_RNA < 50000 & percent.mt < 40) # This gives us 10,363 samples after filtering
seu_sub <- NormalizeData(seu_sub, normalization.method = "LogNormalize", scale.factor = 10000)
seu_sub <- FindVariableFeatures(seu_sub, selection.method = "vst", nfeatures = 3000)
seu_sub <- ScaleData(seu_sub)
seu_sub <- RunPCA(seu_sub, features = VariableFeatures(object = seu_sub), npcs = 50)
#ElbowPlot(seu_sub, 50)
seu_sub <- FindNeighbors(seu_sub, dims = 1:40)
seu_sub <- FindClusters(seu_sub, resolution = 0.5)
seu_sub <- RunUMAP(seu_sub, dims = 1:40)

## Data output for QC ##
DimPlot(seu_sub)
VlnPlot(seu_sub, features = c("nFeature_RNA", "log_RNA", "logUMI", "percent.mt"), group.by = "cart", ncol = 4, pt.size = 0)
DimPlot(seu_sub, group.by = "cart")
FeaturePlot_scCustom(seu_sub, features = "percent.mt")
#####

## Normalize ADT Data ##
DefaultAssay(seu_sub)<-"ADT"
seu_sub <- NormalizeData(seu_sub, normalization.method = "CLR")
seu_sub <- ScaleData(seu_sub)

DimPlot(seu_sub, cols = clus_cols36)


#### Now choose only the ISC cells #####
ab<-"CD326"
dat<-as.data.frame(seu_sub@assays$ADT@data[ab,])
colnames(dat)<-"value"
ggplot(dat, aes(x=value))+geom_density()
seu_sub@meta.data$Ab1<-seu_sub@assays$ADT@data[ab,]>1

ab<-"CD326:G8.8-AMM2281"
dat<-as.data.frame(seu_sub@assays$ADT@data[ab,])
colnames(dat)<-"value"
ggplot(dat, aes(x=value))+geom_density()
seu_sub@meta.data$Ab2<-seu_sub@assays$ADT@data[ab,]>1

ab<-"CD326:G8.8-AMM2294"
dat<-as.data.frame(seu_sub@assays$ADT@data[ab,])
colnames(dat)<-"value"
ggplot(dat, aes(x=value))+geom_density()
seu_sub@meta.data$Ab3<-seu_sub@assays$ADT@data[ab,]>1

ab<-"CD326:G8.8-AMM2295"
dat<-as.data.frame(seu_sub@assays$ADT@data[ab,])
colnames(dat)<-"value"
ggplot(dat, aes(x=value))+geom_density()
seu_sub@meta.data$Ab4<-seu_sub@assays$ADT@data[ab,]>1

seu_sub@meta.data$multiplet<-rowSums(seu_sub@meta.data[,colnames(seu_sub@meta.data) %in% paste0("Ab", 1:4)]+1-1)>1
seu_sub@meta.data$Ab_not_found<-rowSums(seu_sub@meta.data[,colnames(seu_sub@meta.data) %in% paste0("Ab", 1:4)]+1-1)<1

DimPlot(seu_sub, group.by = "multiplet")
DimPlot(seu_sub, group.by = "Ab_not_found")

seu_sub<-seu_sub[,!seu_sub$multiplet & !seu_sub$Ab_not_found] # Now we are down to 9,135 cells from 10,363 cells
#####
seu_sub$group<-c("Female_IFNGRpos", "Female_IFNGRneg", "Male_IFNGRpos", "Male_IFNGRneg")[apply(seu_sub@meta.data[,colnames(seu_sub@meta.data) %in% paste0("Ab", 1:4)], 1, which.max)]

DimPlot(seu_sub, group.by = "group")

#### Temporary Save Point Here ####
#The RDS file below has removed multiplets / those without proper hashtags, as well as filter by RNA counts and mito genes.  
#Also, 4 groups (female IFNy +/- and male IFNy +/-) have been added
saveRDS(seu_sub, file.path(CDS_DIR, "241104_processed.RDS"))
####
```

```{r - modification for DBEC - gives essentially the same outcomes as RSEC so will just use RSEC, echo=FALSE, include=FALSE}

## this uses DBEC
f<-list.files(file.path(DATA_DIR, paste0("cart", 1:2)), full.names = T)[grep("DBEC_MolsPerCell", list.files(file.path(DATA_DIR, paste0("cart", 1:2))))] #This line lists all files in two directories, cart1 and cart2, inside DATA_DIR.
files<-f[c(2,4)]

dat<-lapply(files, function(file){
t(as.matrix(data.table::fread(file), rownames=1))}) # For each file in files, it reads the file into a matrix using data.table::fread. The t() function transposes each matrix, making rows into columns and columns into rows. dat will contain two transposed matrices from the selected files.

rn<-intersect(rownames(dat[[1]]), rownames(dat[[2]])) # rn is a list of row names common to both matrices.
rn<-sapply(strsplit(rn, "\\|"), "[[", 1)  # The strsplit(rn, "\\|") splits each row name by the "|" delimiter. sapply(..., "[[", 1) extracts the first part before the delimiter for each name.

dat2<-lapply(dat, function(mat){
  rownames(mat)<-sapply(strsplit(rownames(mat), "\\|"), "[[", 1)
  mat[rn,]
}) #This modifies the row names of each matrix by keeping only the first part (before the "|"). Each matrix is then subsetted to include only the rows that match the names in rn, creating a new list dat2 with the processed matrices.

colnames(dat2[[1]])<-paste0("cart1_", colnames(dat2[[1]]))
colnames(dat2[[2]])<-paste0("cart2_", colnames(dat2[[2]]))
mat<-Matrix(do.call(cbind, dat2), sparse=T) # The two matrices in dat2 are combined column-wise (i.e., bound together as columns) using do.call(cbind, dat2). The Matrix() function with sparse=T creates a sparse matrix from the combined data, saving memory by storing only non-zero entries.
colnames(mat)<-make.unique(colnames(mat)) # This line ensures that all column names in the combined matrix mat are unique by appending numbers where duplicate names are found.


# meta<-DataFrame(cell.names=colnames(mat), sample_no=unlist(lapply(1:8, function(x) {
#   rep(names(folders)[[x]], ncol(dat[[x]]))})), row.names = colnames(mat))
# meta$sample <- factor(meta$sample_no)
# levels(meta$sample)<-c(rep("Spl", 2),rep("mLN", 2),rep("LP", 2),rep("EL", 2))
# genes[[1]]

rmat<-mat[8:dim(mat)[1],] #rmat contains the main data starting from the 8th row.
pmat<-mat[1:7,] #pmat contains the first 7 rows, potentially holding different or specific information.

#genes<-DataFrame(id = rn,  gene_short_name=rn, row.names = rn)
#rownames(mat)


seu_DBEC<-CreateSeuratObject(counts = rmat)
seu_DBEC$ADT<-CreateAssayObject(counts = pmat[,match(colnames(pmat), Cells(seu_DBEC))])
#DefaultAssay(seu_DBEC)

seu_DBEC <- NormalizeData(seu_DBEC, normalization.method = "LogNormalize", scale.factor = 10000)
seu_DBEC <- FindVariableFeatures(seu_DBEC, selection.method = "vst", nfeatures = 3000)
seu_DBEC <- ScaleData(seu_DBEC)
seu_DBEC <- RunPCA(seu_DBEC, features = VariableFeatures(object = seu_DBEC), npcs = 50)
#ElbowPlot(seu_DBEC, 50)
seu_DBEC <- FindNeighbors(seu_DBEC, dims = 1:40)
seu_DBEC <- FindClusters(seu_DBEC, resolution = 0.4)
seu_DBEC <- RunUMAP(seu_DBEC, dims = 1:40)

DimPlot(seu_DBEC)

seu_DBEC$cart<-strsplit(Cells(seu_DBEC), "_") %>% sapply("[[", 1)
seu_DBEC[["percent.mt"]] <- PercentageFeatureSet(seu_DBEC, pattern = "^mt-")
seu_DBEC$logUMI<-log10(seu_DBEC$nCount_RNA)
seu_DBEC$log_RNA<-log10(seu_DBEC$nCount_RNA)


## For QC presentation ##
VlnPlot(seu_DBEC, features = c("nFeature_RNA", "log_RNA", "logUMI", "percent.mt"), group.by = "cart", ncol = 4, pt.size = 0)
DimPlot(seu_DBEC, group.by = "cart")
FeaturePlot_scCustom(seu_DBEC, features = "percent.mt")
####

#DefaultAssay(seu)<-"RNA"
#add percent.mt to metadata
#seu$percent.mt <- PercentageFeatureSet(seu, pattern = "^mt-") #This is the percentage of mitochondrial genes (low-quality / dying cells often have higher %; common cutoff is 5)

# Visualize QC metrics as a violin plot
#VlnPlot(seu, features = c("log_RNA"), pt.size = 0)
#VlnPlot(seu, features  =c("percent.mt"), pt.size = 0) #+ylim(c(0,15))
#FeatureScatter(seu, feature1 = "nCount_RNA", feature2 = "percent.mt") # Simple scatter function

#subset RNA data, these are  standard parameters
#seu <- subset(seu, subset = log_RNA > 3.25 & log_RNA < 4.25 & percent.mt < 5)

```

```{r - isolate ISCs, echo=FALSE, include=TRUE}
seu<-readRDS(file.path(CDS_DIR, "241104_processed.RDS"))

# [1] "CD326:G8.8-AMM2281" "CD326:G8.8-AMM2294" "CD326:G8.8-AMM2295" "CD326"             
# [5] "CD45.2"             "H-2Kb"              "I-A-I-E"  

# Now, we select the group that has ISCs, as denoted by Lgr5+

DefaultAssay(seu)<-"ADT"

# Generate the DimPlot and then add custom labels

#---- For Supp Fig 5 ---#
pdf(file.path(FIG_DIR, paste0("[Final] - QC Clusters", ".pdf")), width=6, height=6)
DimPlot(seu, label = TRUE, label.size = 5, repel = TRUE, label.box = TRUE) + scale_fill_manual(values = rep("white",12))  # Ensure label boxes are white
dev.off()
#---- For Supp Fig 5 ---#

DefaultAssay(seu)<-"RNA"

#---- For Supp Fig 5 ---#
pdf(file.path(FIG_DIR, paste0("[Final] - QC Lgr5", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Lgr5")
dev.off()
#---- For Supp Fig 5 ---#

VlnPlot(seu, features = "Lgr5", pt.size = 0.1) 
RidgePlot(seu, features = "Lgr5", ncol = 2)
DotPlot(seu, features = "Lgr5") + RotatedAxis()

#---- For Supp Fig 5 ---#
my_comparisons <- list( c("5", "8"), c("6", "8"), c("9", "8"))
pdf(file.path(FIG_DIR, paste0("[Final] - QC Lgr5 Exp Level", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("Lgr5"), pt.size = 0) + stat_summary(geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()
#---- For Supp Fig 5 ---#


VlnPlot(seu, features = c("Lgr5"), pt.size = 0)+ stat_summary(fun = "mean", geom='point', size = 25, colour = "black", shape = 95) + ylim(0, 7)



markers_7_vs_8 <- FindMarkers(seu, ident.1 = 7, ident.2 = 8) #Helped identify Muc2 (clusters 5,6,8) = goblet cells

#---- For Supp Fig 5 ---#
pdf(file.path(FIG_DIR, paste0("[Final] - QC Muc2", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Muc2")
dev.off()
#---- For Supp Fig 5 ---#

#---- For Supp Fig 5 ---#
pdf(file.path(FIG_DIR, paste0("[Final] - QC Trpm5", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Trpm5")
dev.off()
#---- For Supp Fig 5 ---#

DotPlot(seu, features = c("Muc2","Lgr5")) + RotatedAxis()

markers_4_vs_10 <- FindMarkers(seu, ident.1 = 4, ident.2 = 10) #Helped identify Trpm5 (cluster 10) = tuft cells
DotPlot(seu, features = c("Lgr5","Muc2","Trpm5")) + RotatedAxis()


# Choose Lgr5+ clusters, which includes 0,1,2,3,4,7,11 (these are higher in Lgr5 and also low in Muc2 (goblet cells), Trpm5 (tuft cells)
seu<-seu[,seu$seurat_clusters %in% c(0,1,2,3,4,7,11)]

DefaultAssay(seu)<-"RNA"
seu <- NormalizeData(seu, normalization.method = "LogNormalize", scale.factor = 10000)
seu <- FindVariableFeatures(seu, selection.method = "vst", nfeatures = 3000)
seu <- ScaleData(seu)
seu <- RunPCA(seu, features = VariableFeatures(object = seu), npcs = 50)

ElbowPlot(seu, 50)
seu <- FindNeighbors(seu, dims = 1:25)
seu <- FindClusters(seu, resolution = 0.5)
seu <- RunUMAP(seu, dims = 1:25)
DimPlot_scCustom(seu, group.by = "group", colors_use = clus_cols36, pt.size = 1) 

DimPlot_scCustom(seu, colors_use = clus_cols36, pt.size = 1, label = TRUE, label.size = 5, repel = TRUE, label.box = TRUE) + scale_fill_manual(values = rep("white",12)) 

#### Temporary Save Point Here ####
#In addition to the filtering from 241104_processed.RDS, this version also removes tuft cells, goblet cells
saveRDS(seu, file.path(CDS_DIR, "241105_processed.RDS"))
####
```

```{r - analysis of ISCs, echo=FALSE, include=TRUE}
seu<-readRDS(file.path(CDS_DIR, "241105_processed.RDS"))
FeaturePlot_scCustom(seu, features = "Reg3b", pt.size = 1)
DimPlot_scCustom(seu, group.by = "group", colors_use = clus_cols36, pt.size = 1) 

FeaturePlot_scCustom(seu, features = "Reg3b", pt.size = 1)
DefaultAssay(seu)<-"ADT"
FeaturePlot_scCustom(seu, features = "I-A-I-E", pt.size = 1)


```



##----- GSEA Analysis 11/5/24 -----##
```{r - fsgea function; run this before GSEA analysis, echo=FALSE, include=TRUE}
plot_fsgeaTEST_bargraph<-function(genedb, genelist_wilcox, cluster, dbname, NES_cutoff, q_value, height=10, width=10){
#This is the fsgea we use for final analysis.  The mutate AUC-0.5 makes the comparisons symmetric
#The NES cutoff plots only gene sets above a certain NES
  set.seed(1234)
  #print(paste0("q-value: ",q_value))
  #q_value<-0.2
  #genedb<- m_t2g
  #genelist_wilcox <- seu.genes46
  #cluster <- c("4")
  #head(genelist_wilcox)
  #dplyr::count(genelist_wilcox,group)
  
  fgsea_sets<- genedb %>% split(x = .$gene_symbol, f = .$gs_name) #Collects names of the genes of interest
  
  #genes45<- genelist_wilcox %>%
    #mutate(auc = auc - 0.5) %>%
    #group_by(feature) %>% # should be 2 rows per feature (group 4 and group 5)
    #mutate(adj = rnorm(n = 1, sd = 0.0001), auc = if_else(group == "4", auc + adj, auc - adj)) %>%
    #arrange(group, desc(auc)) %>%
    #dplyr::select(group, feature, auc)
    
  #genes <- genes45 %>% filter(group == cluster) %>% dplyr::select(-group)
  genes <- genelist_wilcox %>%
    mutate(auc = auc - 0.5) %>%
    group_by(feature) %>%
#    mutate(adj = rnorm(n = 1, sd = 0.0001), auc = if_else(group == cluster, auc + adj, auc - adj)) %>%
    dplyr::filter(group == cluster) %>%
    arrange(desc(auc)) %>% 
    dplyr::select(feature, auc) 
  
  ranks<- deframe(genes) #Restructures based on tibble package
  #ranks[ranks>0] <- 2*ranks[ranks>0]

  fgseaRes<- fgsea(fgsea_sets, stats = ranks, nperm = 1000000, nproc=8)
  
  fgseaResTidy <- fgseaRes %>%
    as_tibble() %>%
    arrange(desc(NES))
  results <- do.call(cbind, fgseaResTidy[,1:5])
  
  smallest<- order(fgseaResTidy$NES)<11 & fgseaResTidy$pval <0.05 #smallest NES with p-values < 0.05
  largest <- order(fgseaResTidy$NES)>max(order(fgseaResTidy$NES))-10 & fgseaResTidy$pval <0.05 #largest NES with p-values < 0.05

  write.csv(results, file.path(FIG_DIR, paste0("[Final] GSEA - ", dbname, " - Group ", cluster, ".csv")))
  fgseaResTidy$pathway<-gsub("HALLMARK_","",fgseaResTidy$pathway)
  
  pdf(file.path(FIG_DIR, paste0("[Final] - GSEA_Enrichment" , dbname,  "_Gp", cluster," ALL sigp with NES cutoff_", NES_cutoff, ".pdf")), width, height)
  print(ggplot(fgseaResTidy[fgseaResTidy$padj<q_value & abs(fgseaResTidy$NES)>NES_cutoff,], aes(reorder(pathway, NES), NES)) + geom_col(aes(fill= ES > 0)) + coord_flip() + labs(x="Pathway", y="Normalized Enrichment Score", title=paste0("Pathways with q<",q_value, "with NES cutoff: ", NES_cutoff)) + theme_classic())
  dev.off()
  return(ranks)
}
```


##----- Other Paper Figures 11/5/24 -----##
```{r Figure 6 Paper}


#----- FOR PUBLICATION / MOTOKO -----#
#----Other custom UMAP plots requested by Motoko ----#

pdf(file.path(FIG_DIR, paste0("[Final] - Clusters", ".pdf")), width=6, height=6)
DimPlot_scCustom(seu, colors_use = clus_cols36, pt.size = 1.5, label = TRUE, label.size = 8, repel = FALSE, label.box = TRUE) + scale_fill_manual(values = rep("white",12)) 
dev.off()
pdf(file.path(FIG_DIR, paste0("[Final] - Groups", ".pdf")), width=7, height=6)
DimPlot_scCustom(seu, group.by = "group", colors_use = clus_cols36, pt.size = 1.5) 
dev.off()


DefaultAssay(seu)<-"RNA"
pdf(file.path(FIG_DIR, paste0("[Final] - Ciita", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Ciita", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H2-Aa", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "H2-Aa", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H2-Ab1", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "H2-Ab1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - Mki67", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "Mki67", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()


### Protein Plots #####
DefaultAssay(seu) <- "ADT"
rownames(seu@assays$ADT)
pdf(file.path(FIG_DIR, paste0("[Final] - I-A-I-E", ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "I-A-I-E", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()



my_comparisons <- list( c("4", "5"), c("5", "6"), c("4", "6"))
pdf(file.path(FIG_DIR, paste0("[Final] - Ciita Exp Level", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("Ciita"), pt.size = 0, cols = clus_cols36)+ stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H2-Aa Exp Level", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("H2-Aa"), pt.size = 0, cols = clus_cols36)+ stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H2-Ab1 Exp Level", ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("H2-Ab1"), pt.size = 0, cols = clus_cols36)+ stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - I-A-I-E Exp Level", ".pdf")), width=6, height=6)
DefaultAssay(seu) <- "ADT"
VlnPlot(seu, features = c("I-A-I-E"), pt.size = 0, cols = clus_cols36)+ stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

```

```{r Supp Figure 1 Paper}
### Comparison by groups ###
my_comparisons <- list( c("Female_IFNGRneg", "Female_IFNGRpos"), 
                        c("Male_IFNGRneg", "Male_IFNGRpos"), 
                        c("Female_IFNGRneg", "Male_IFNGRneg"), 
                        c("Female_IFNGRpos","Male_IFNGRpos"), 
                        c("Female_IFNGRneg","Male_IFNGRpos"), 
                        c("Female_IFNGRpos","Male_IFNGRneg"))

pdf(file.path(FIG_DIR, paste0("[Final] - Ciita (by group)" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("Ciita"), group.by="group", pt.size = 0) + stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H2-Aa (by group)" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("H2-Aa"), group.by="group", pt.size = 0) + stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - H2-Ab1 (by group)" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("H2-Ab1"), group.by="group", pt.size = 0) + stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - I-A-I-E (by group)" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("I-A-I-E"), group.by="group", pt.size = 0) + stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 7)
dev.off()



m_t2g_C <- msigdbr(species = "Mus musculus", category = "C5", subcategory = "GO:CC") %>% dplyr::select(gs_name, gene_symbol) 
bac<-m_t2g_C[m_t2g_C$gs_name == "GOCC_MHC_CLASS_II_PROTEIN_COMPLEX",]$gene_symbol 
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "GOCC_MHC_CLASSII")

pdf(file.path(FIG_DIR, paste0("[Final] - GOCC_MHC_CLASS_II_PROTEIN_COMPLEX Plot" , ".pdf")), width=6, height=6)
FeaturePlot_scCustom(seu, features = "GOCC_MHC_CLASSII1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

pdf(file.path(FIG_DIR, paste0("[Final] - GOCC_MHC_CLASS_II_PROTEIN_COMPLEX Violin" , ".pdf")), width=6, height=6)
VlnPlot(seu, features = c("GOCC_MHC_CLASSII1"), group.by="group", pt.size = 0) + stat_summary(fun.y = median, geom='point', size = 10, colour = "black", shape = 95) + stat_compare_means(comparisons = my_comparisons, method = "wilcox.test") + ylim(0, 4)
dev.off()



```

```{r GSEA Figures / Supp Figure 2 Paper}

#========== Figure 5E GSEA ==========#
#---- GSEA analysis using fsgea ----#
#https://crazyhottommy.github.io/scRNA-seq-workshop-Fall-2019/scRNAseq_workshop_3.html
library(presto)
library(dplyr)
library(ggplot2)
library(tibble)
library(fgsea)
library(msigdbr)

seu.genes <- wilcoxauc(seu, 'seurat_clusters')
seu.genes2 <- wilcoxauc(seu,'group')

# for GSEA, we need the information of all genes, Seurat is just too slow if we test
# all 20,000 genes. instead let's try presto which performs a fast Wilcoxon rank sum test 

#---- Hallmark Gene Sets ----#
m_t2g <- msigdbr(species = "Mus musculus", category = "H") %>% dplyr::select(gs_name, gene_symbol) 
fgsea_sets<- m_t2g %>% split(x = .$gene_symbol, f = .$gs_name) #Collects names of the genes of interest
#----- Comparing between groups -----#


#----- FOR PUBLICATION / MOTOKO -----#

seu.genes45 <- wilcoxauc(seu,"seurat_clusters",groups_use=c('4','5'))
# Mutate with AUC = AUC-0.5, specify q-value
fsgea_45_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes45,"4","Hallmark Set 4v5 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)
fsgea_45_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes45,"5","Hallmark Set 4v5 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)

seu.genes46 <- wilcoxauc(seu,"seurat_clusters",groups_use=c('4','6'))
# Mutate with AUC = AUC-0.5, specify q-value
fsgea_46_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes46,"4","Hallmark Set 4v6 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)
fsgea_46_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes46,"6","Hallmark Set 4v6 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)


#---- Used in Figure 6 ----#
seu.genes56 <- wilcoxauc(seu,"seurat_clusters",groups_use=c('5','6'))
fsgea_56_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes56,"5","Hallmark Set 5v6 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)
fsgea_56_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes56,"6","Hallmark Set 5v6 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)
#---- Used in Figure 6 ----#

seu.genes456 <- wilcoxauc(seu,"seurat_clusters",groups_use=c('4','5','6'))
fsgea_456_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes456,"4","Hallmark Set 4v5_6 Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)


pdf(file.path(FIG_DIR, paste0("[Final] - H_IFN1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_INTERFERON_GAMMA_RESPONSE",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_IFN")
FeaturePlot_scCustom(seu, features = "H_IFN1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

## Plot Groups individually 
seu$group_Male_IFNGRneg <- seu$group=="Male_IFNGRneg"
pdf(file.path(FIG_DIR, paste0("[Final] - group_Male_IFNGRneg", ".pdf")), width=6, height=6)
seu_gray <- subset(seu, subset = group_Male_IFNGRneg == "TRUE")
DimPlot_scCustom(seu_gray, group.by = "group_Male_IFNGRneg", colors_use = "Gray", pt.size = 1)
dev.off()

seu$group_Female_IFNGRneg <- seu$group=="Female_IFNGRneg"
pdf(file.path(FIG_DIR, paste0("[Final] - group_Female_IFNGRneg", ".pdf")), width=6, height=6)
seu_gray <- subset(seu, subset = group_Female_IFNGRneg == "TRUE")
DimPlot_scCustom(seu_gray, group.by = "group_Female_IFNGRneg", colors_use = "Gray", pt.size = 1)
dev.off()

seu$group_Male_IFNGRpos <- seu$group=="Male_IFNGRpos"
pdf(file.path(FIG_DIR, paste0("[Final] - group_Male_IFNGRpos", ".pdf")), width=6, height=6)
seu_gray <- subset(seu, subset = group_Male_IFNGRpos == "TRUE")
DimPlot_scCustom(seu_gray, group.by = "group_Male_IFNGRpos", colors_use = "Gray", pt.size = 1)
dev.off()

seu$group_Female_IFNGRpos <- seu$group=="Female_IFNGRpos"
pdf(file.path(FIG_DIR, paste0("[Final] - group_Female_IFNGRpos", ".pdf")), width=6, height=6)
seu_gray <- subset(seu, subset = group_Female_IFNGRpos == "TRUE")
DimPlot_scCustom(seu_gray, group.by = "group_Female_IFNGRpos", colors_use = "Gray", pt.size = 1)
dev.off()

#---- Used in Figure 6 ----#
seu.genes_groups <- wilcoxauc(seu,'group', groups_use=c('Male_IFNGRpos','Male_IFNGRneg'))
    # Mutate with AUC = AUC-0.5
fsgea_groups_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes_groups,"Male_IFNGRpos","Hallmark Set MvM IFNGRneg v pos Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)

fsgea_groups_ranks <- plot_fsgeaTEST_bargraph(m_t2g,seu.genes_groups,"Male_IFNGRneg","Hallmark Set MvM IFNGRpos v neg Mutate AUC, q_0.2",0, q_value=0.2, height=6, width=6)
#---- Used in Figure 6 ----#

    
#----- FOR PUBLICATION / MOTOKO -----#
```

```{r Supp Figure 3 Paper}


#------ Plot gene signatures onto map ------#

pdf(file.path(FIG_DIR, paste0("[Final] - H_OXPHOS1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_OXIDATIVE_PHOSPHORYLATION",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_OXPHOS")
FeaturePlot_scCustom(seu, features = "H_OXPHOS1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_E2F1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_E2F_TARGETS",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_E2F")
FeaturePlot_scCustom(seu, features = "H_E2F1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_KRAS_UP1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_KRAS_SIGNALING_UP",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_KRAS_UP")
FeaturePlot_scCustom(seu, features = "H_KRAS_UP1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)



pdf(file.path(FIG_DIR, paste0("[Final] - H_KRAS_DOWN1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_KRAS_SIGNALING_DN",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_KRAS_DOWN")
FeaturePlot_scCustom(seu, features = "H_KRAS_DOWN1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)




pdf(file.path(FIG_DIR, paste0("[Final] - H_IFN1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_INTERFERON_GAMMA_RESPONSE",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_IFN")
FeaturePlot_scCustom(seu, features = "H_IFN1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)

pdf(file.path(FIG_DIR, paste0("[Final] - H_GLYCOLYSIS1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_GLYCOLYSIS",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_GLYCOLYSIS")
FeaturePlot_scCustom(seu, features = "H_GLYCOLYSIS1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)



pdf(file.path(FIG_DIR, paste0("[Final] - H_MITO1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_MITOTIC_SPINDLE",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_MITO")
FeaturePlot_scCustom(seu, features = "H_MITO1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_TNF1", ".pdf")), width=6, height=6)
rna_cols<-paletteContinuous(n=8)[c(1:3, 6:8)]
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_TNFA_SIGNALING_VIA_NFKB",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_TNF")
FeaturePlot_scCustom(seu, features = "H_TNF1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_HEDGEHOG1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_HEDGEHOG_SIGNALING",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_HEDGEHOG")
FeaturePlot_scCustom(seu, features = "H_HEDGEHOG1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_IFNA1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_INTERFERON_ALPHA_RESPONSE",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_IFNA")
FeaturePlot_scCustom(seu, features = "H_IFNA1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_FAT1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_FATTY_ACID_METABOLISM",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_FAT")
FeaturePlot_scCustom(seu, features = "H_FAT1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_G2M1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_G2M_CHECKPOINT",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_G2M")
FeaturePlot_scCustom(seu, features = "H_G2M1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_WNT1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_WNT_BETA_CATENIN_SIGNALING",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_WNT")
FeaturePlot_scCustom(seu, features = "H_WNT1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)


pdf(file.path(FIG_DIR, paste0("[Final] - H_TGFB1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_TGF_BETA_SIGNALING",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_TGFB")
FeaturePlot_scCustom(seu, features = "H_TGFB1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()

sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)



pdf(file.path(FIG_DIR, paste0("[Final] - H_GLYCOLYSIS1", ".pdf")), width=6, height=6)
bac<-m_t2g[m_t2g$gs_name == "HALLMARK_MYC_TARGETS_V2",]$gene_symbol
DefaultAssay(seu)<-"RNA"
seu<-AddModuleScore(seu, features = list(bac), name = "H_GLYCOLYSIS")
FeaturePlot_scCustom(seu, features = "H_GLYCOLYSIS1", label = F, label.size = 2, pt.size = 2, max.cutoff = "q99")+scale_color_gradientn(colors = rna_cols)&NoAxes()
dev.off()
sum(rownames(seu) %in% bac)
length(bac)
sum(rownames(seu) %in% bac)/length(bac)



```
